cluster-autoscaler/utils/drain/drain.go

```diff
@@ -231,6 +158,104 @@ func GetPodsForDeletionOnNodeDrain(
 	return pods, daemonSetPods, nil, nil
 }
 
+func legacyCheckForReplicatedPods(listers kube_util.ListerRegistry, pod *apiv1.Pod, minReplica int32) (replicated bool, isDaemonSetPod bool, blockingPod *BlockingPod, err error) {
```

Comment:
The name `legacyCheckForReplicatedPods` doesn't account for checking for daemon set pods or blocking pods but using `legacyCheckForReplicatedAndDaemonSetAndBlockingPods` feels too long so I have kept it as is for now.

Reply:
Note that the return type `replicated bool, isDaemonSetPod bool, blockingPod *BlockingPod, err error` has names. I did this because I think it improves readability when you hover over the function call like in IDEs e.g.,
ile
![image](https://user-images.githubusercontent.com/34534103/225218854-883e4468-9c25-47ea-9b28-32c43a08ec09.png)

This might not be the best approach since it can confuse someone when they look at line 162 and 165 below (because the variables are already defined due to them being named return types)